lib/hiera/backend/compliance_markup_backend.rb

```diff
@@ -0,0 +1,104 @@
+# vim: set expandtab ts=2 sw=2:
+class Hiera
+  module Backend
+    class Compliance_markup_backend
+      def initialize
+        # 
+        # Load the shared compliance_mapper codebase
+        #
+        filename = File.dirname(File.dirname(File.dirname(__FILE__))) + "/puppetx/simp/compliance_mapper.rb"
+        self.instance_eval(File.read(filename), filename)
+
+        # Grab the config from hiera
+        @config = Config[:compliance_markup]
+
+        @cache_timeout = @config[:cache_timeout] || 600
+        @cache_clean_interval = @config[:cache_clean_interval] || 3600
+        @cache_last_check = Time.now.to_i + @cache_clean_interval
+      end
+
+      def lookup(key, scope, order_override, resolution_type, context)
+        # Monkey patch the catalog *object* to add a _compliance_cache accessor
+        # We do this to prevent environment poisoning by monkey patching the class,
+        # and it still allows us to have a catalog scoped cache.
+
+        begin
+          @cache = scope.catalog._compliance_cache
+        rescue
+          scope.catalog.instance_eval("
+           def _compliance_cache=(value)
+             @_compliance_cache = value
+           end
+           def _compliance_cache
+             @_compliance_cache
+           end")
+          scope.catalog._compliance_cache = {}
+          @cache = scope.catalog._compliance_cache
+        end
+
+
+        now = Time.now.to_i
+        if (now > @cache_last_check)
+          clean_cache(now)
```

Comment:
I'm actually going to rip that out. It isn't necessary since we explicitly cache per catalog now